src/node/in_memory.rs

```diff
@@ -1350,19 +1359,17 @@ impl<S: ForkSource + std::fmt::Debug + Clone> InMemoryNode<S> {
         let call_tracer_result = Arc::new(OnceCell::default());
         let bootloader_debug_result = Arc::new(OnceCell::default());
 
-        tracers.push(CallErrorTracer::new().into_tracer_pointer());
-        tracers.push(CallTracer::new(call_tracer_result.clone()).into_tracer_pointer());
-        tracers.push(
+        let tracers = vec![
+            CallErrorTracer::new().into_tracer_pointer(),
+            CallTracer::new(call_tracer_result.clone()).into_tracer_pointer(),
             BootloaderDebugTracer {
                 result: bootloader_debug_result.clone(),
             }
             .into_tracer_pointer(),
-        );
-
+        ];
         let (compressed_bytecodes, tx_result) =
             vm.inspect_transaction_with_bytecode_compression(&mut tracers.into(), tx.clone(), true);
```

Comment:
Using `forge script` we get a failure due to account validation error that seems to be from here.

```
Failed to send transaction

Context:
- server returned an error response: error code 3: Execution error: Transaction HALT: Account validation error: Not enough gas for transaction validation, data: "0x9b0025a197e7157cd45dc5658a2f6748ddfc87c25d5ad00a9a29d34de2dcc7a8"
```

This is when using the rich wallets from era-test-node and default Counter script from Foundry-rs:

```
forge script script/Counter.s.sol:CounterScript --rpc-url http://127.0.0.1:8011 --broadcast --sender 0x87d6ab9fE5Adef46228fB490810f0F5CB16D6d04 --private-key <RICH-WALLET-PK> -vvvv
```

Reply:
The out-of-gas error is exactly what it is (i.e., it isn't misinterpreted or masks another error). It's caused by extremely high gas per pubdata byte value (50,000 gas / byte) inferred from the default fee params. This has nothing to do with this PR; I've checked that the value is the same on the `main` branch. AFAIU, `forge` doesn't take it into account at all when not running in the `--zksync` mode, so it sets gas limit at ~130k. However, the transaction actually publishes lots of pubdata (664 bytes in my local run – it may be caused by the EVM emulator, but not entirely sure), which, combined with the aforementioned high price, leads to the gas usage of ~33.5M. If you adjust gas limit correspondingly by specifying something like `-g 50000` (so 500x the estimated value 🙃), the transaction is accepted.

Questions:

- Maybe, we want to decrease the default gas per pubdata byte value by adjusting fee params? AFAIU, it's much lower on the mainnet etc.
- @0xVolosnikov Can you confirm whether pubdata published by the EVM emulator is a reasonable hypothesis? If so, how would it factor into DevEx given that (AFAIU) there's no equivalent of it on Ethereum, so tools like `forge` don't take it into account? Maybe it should be subsidized in some way inside the bootloader, and it's just not implemented yet?

Reply:
@slowli 
> Can you confirm whether pubdata published by the EVM emulator is a reasonable hypothesis

Yes, this transaction deploys the smart contract and publishes its bytecode on L1. But ~33.5M is really huge indeed.  

> so tools like forge don't take it into account

These tools should use the `eth_estimateGas`. Do we take into account the pubdata during gas estimation?

Reply:
> These tools should use the `eth_estimateGas`.

They don't necessarily though. I don't know where `forge` gets its 130k gas estimation from, but it doesn't call `eth_estimateGas` at all before sending the transaction, only `eth_gasPrice`. `forge create` does use `eth_estimateGas`, so it works without specifying a gas limit multiplier (naturally, gas limits it gets are still quite high).